app/models/publication.rb

```diff
@@ -250,31 +250,23 @@ def authoritative_pmid_source?
 
   private
 
-    def add_all_identifiers_in_db_to_pub_hash
-      publication_identifiers.reload if persisted?
-      pub_hash[:identifier] = publication_identifiers.map(&:identifier)
-    end
-
-    def sync_identifiers_in_pub_hash_to_db
+    # doesn't actually write the Pub to DB, presumed to be part of before_save callback or explicit save
+    def sync_identifiers_in_pub_hash
       incoming_types = Array(pub_hash[:identifier]).map { |id| id[:type] }
       publication_identifiers.each do |id|
         next if id.identifier_type =~ /^legacy_cap_pub_id$/i # Do not delete legacy_cap_pub_id
-        id.delete unless incoming_types.include? id.identifier_type
+        publication_identifiers.delete(id) unless incoming_types.include?(id.identifier_type)
       end
```

Comment:
Not entirely clear about what class we get from `publication_identifiers` (I guess it is an enumerable of `PublicationIdentifier`).  This change seems to remove the AR-record from the enumerable, but not from the db (if I understand this right).  If that's the intention, maybe `publication_identifiers.reject!` declares it up front and simplifies the block body.  Seems like a big change not to delete the db-record though, need to wrap my head around this when I've had some :coffee:

Reply:
Yes, you are following the right thread, but missing some context.  It removes it from the association, and then it takes *whatever action* is prescribed by the association itself.  In this case it is declared with 
```ruby
dependent: :destroy
```
Another option, `:nullify`, would keep the record. But this one should destroy it.

Reply:
```ruby
publication_identifiers.select {|id| id.identifier_type !~ /^legacy_cap_pub_id$/i }.reject do |id|
  ! incoming_types.include?(id.identifier_type)
end
```